src/terrain/terrain.cpp

```diff
@@ -439,28 +438,32 @@ Terrain::init_grass() {
     for (int x_ = 0; x_ < X_MAX; x_++)
         for (int y_ = 0; y_ < Y_MAX; y_++) {
             get_tile(x_, y_, z_)->try_grow_grass(); // add to sources and sinks
-            if (get_tile(x_, y_, z_)->is_grass()){
+            if (get_tile(x_, y_, z_)->is_grass()) {
                 all_grass.insert(get_tile(x_, y_, z_));
             }
             // same thing here as above
         }
     grow_grass_high(all_grass);
     grow_grass_low(all_grass);
     for (Tile* t : all_grass) {
-        t->set_grass_color(grass_grad_length, grass_mid, grass_colors);
+        t->set_grass_color(grass_grad_length_, grass_mid_, grass_colors_);
     }
 }
 
 void
-Terrain::grow_grass_high(std::set<Tile*> all_grass){
-    helper::grow_grass_recursive<helper::edge_detector_high,
-        helper::getter_high, helper::setter_high>(*this, all_grass);
+Terrain::grow_grass_high(std::set<Tile*> all_grass) {
+    helper::grow_grass_recursive<
+        helper::edge_detector_high, helper::getter_high, helper::setter_high>(
+        *this, all_grass
+    );
 }
 
 void
-Terrain::grow_grass_low(std::set<Tile*> all_grass){
-    helper::grow_grass_recursive<helper::edge_detector_low, helper::getter_low,
-        helper::setter_low>(*this, all_grass);
+Terrain::grow_grass_low(std::set<Tile*> all_grass) {
+    helper::grow_grass_recursive<
+        helper::edge_detector_low, helper::getter_low, helper::setter_low>(
+        *this, all_grass
+    );
 }
```

Comment:
Put these in a header file and mark them `inline`.

Or if they're private make them `static inline` functions not inside a class.